test/test_torch.py

```diff
@@ -763,6 +763,45 @@ def test_where_bool_tensor(self):
             res = torch.where(a > 0)
             self.assertEqual(1, len(res))
 
+    def test_where_tensor(self):
+        def rand_tensor(size, dtype, device):
```

Comment:
Could we please deduplicate the rand_tensor calls? This is the third place we have logic to create a random tensor; this looks pretty similar to the one in test_type_promotion.

Reply:
as in the other review, there's plenty we could do, but I want master to match 1.3.1 first.

Reply:
this is just a cherry pick from 1.3.1 to master, also need to merge to 1.4

Reply:
I'm pretty uncomfortable with having the same code duplicated. Could you please send a followup PR to deduplicate this and https://github.com/pytorch/pytorch/blob/8b29701ae505a17595a85cac05109ddf3f82e68e/test/test_type_promotion.py#L174-L187 ?

Reply:
there's an issue filed. I added a reference to this in the issue tracker:
https://github.com/pytorch/pytorch/issues/30611